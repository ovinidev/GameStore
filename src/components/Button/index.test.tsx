import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import { Button } from ".";

describe("Button", () => {
	describe("Render button tests", () => {
		it("should bea ble to render correctly with children", () => {
			const { getByText } = render(<Button>Button</Button>);

			expect(getByText("Button")).toBeTruthy();
		});

		it("should be able to handle onPress event", () => {
			const mockPressHandler = jest.fn();
			const { getByText } = render(
				<Button onPress={mockPressHandler}>Test</Button>,
			);

			fireEvent.press(getByText("Test"));

			expect(mockPressHandler).toHaveBeenCalled();
		});

		it("should forward props to TouchableOpacity", () => {
			const { getByLabelText } = render(
				<Button accessibilityLabel="button test">Test</Button>,
			);
			const buttonElement = getByLabelText("button test");

			expect(buttonElement).toBeTruthy();
		});
	});

	describe("Button Snapshots", () => {
		it("should be able to match the snapshot with default props", () => {
			const { toJSON } = render(<Button>Test</Button>);

			expect(toJSON()).toMatchSnapshot();
		});
	});
});
