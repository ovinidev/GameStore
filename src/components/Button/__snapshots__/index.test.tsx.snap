// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button Snapshots should be able to match the snapshot with default props 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <Text
    className="text-center font-bold text-md text-slate-50"
  >
    Test
  </Text>
</View>
`;
